libs/wingc/src/lsp/rename_visitor.rs

```diff
@@ -0,0 +1,167 @@
+use std::sync::Mutex;
+
+use lsp_types::{Position, Range, TextEdit};
+
+use crate::diagnostic::WingLocation;
+use crate::parser::RESERVED_WORDS;
+use crate::type_check::symbol_env::LookupResult;
+use crate::type_check::{SymbolKind, Types};
+use crate::visit::{visit_scope, Visit};
+use crate::visit_context::{VisitContext, VisitorWithContext};
+use crate::{ast::*, visit_context};
+pub struct LinkedSymbol<'a> {
+	symbol: Symbol,
+	references: Mutex<Vec<&'a Symbol>>,
```

Comment:
This shouldn't be needed. All cases where we need to mutate the vector allows us to get a mutable reference to this value.

I wasn't sure about this so I had to try it locally, so hopefully it's okay I just push up the changes 🙏

Reply:
sure! I tested that as well, thanks for the fix 🙏